### PR TITLE
Add `imag` specification for returning the imaginary component of a complex number

### PR DESCRIPTION
**Pull request by kgryte**
This PR

-   adds a specification for `imag`, an API for returning the imaginary component of a complex number for each element ``x_i`` of an input array ``x``.

-   follows https://github.com/data-apis/array-api/pull/427 and https://github.com/data-apis/array-api/pull/446 in restricting the API to only arrays having complex floating-point dtypes. This is as discussed in the Sept 22, 2022, Consortium workgroup meeting.

### CODE REVIEW

**Review by leofang — APPROVED** [review]

**Comment by kgryte** [comment]
As this is a straightforward addition and has already been discussed in consortium meetings, will merge. Any further revisions can be addressed in a subsequent PR.

File: spec/API_specification/array_api/elementwise_functions.py [hunk 1]
```diff
@@ -785,6 +785,21 @@ def greater_equal(x1: array, x2: array, /) -> array:
...
+
+    Parameters
+    ----------
+    x: array
+        input array. Should have a complex floating-point data type.
+
+    Returns
+    -------
+    out: array
+        an array containing the element-wise results. The returned array must have a floating-point data type with the same floating-point precision as ``x`` (e.g., if ``x`` is ``complex64``, the returned array must have the floating-point data type ``float32``).
```
Comment (asmeurer) [inline spec/API_specification/array_api/elementwise_functions.py:800]: Should this be "must" or "should"?

Comment (kgryte) [reply spec/API_specification/array_api/elementwise_functions.py:800]: In what scenarios should/would this not be the case? And why?

Comment (asmeurer) [reply spec/API_specification/array_api/elementwise_functions.py:800]: Oh my bad I was reading this as if it were the input array, not the output. I see the input does say "should".

Comment (kgryte) [reply spec/API_specification/array_api/elementwise_functions.py:800]: For the input dtype, allowing array libraries to support real-valued dtypes makes sense in order to allow for backward compat; however, all array libraries should return an array having a dtype with the same precision.

I suppose you may be thinking if an array library supports real-valued int dtypes, would requiring a float dtype make sense.

**Pull request merged**